tests/refs/reflog/messages.c

```diff
@@ -0,0 +1,408 @@
+
+	cl_git_pass(git_commit_tree(&tree, b1_commit));
+
+	cl_git_pass(git_commit_create(&merge_commit_oid,
+		g_repo, "HEAD", s, s, NULL,
+		"Merge commit", tree,
+		2, (const struct git_commit **) parent_commits));
+
+	reflog_check_entry(g_repo, GIT_HEAD_FILE, 0,
+		NULL,
+		git_oid_tostr_s(&merge_commit_oid),
+		NULL, "commit (merge): Merge commit");
+
+	git_tree_free(tree);
+	git_commit_free(b1_commit);
+	git_commit_free(b2_commit);
+	git_signature_free(s);
+}
+
+void test_refs_reflog_messages__creating_a_direct_reference(void)
+{
+	git_reference *reference;
+	git_oid id;
+	git_reflog *reflog;
+	const git_reflog_entry *entry;
+
+	const char *name = "refs/heads/new-head";
+	const char *message = "You've been logged, mate!";
+
+	cl_git_pass(git_reference_name_to_id(&id, g_repo, "HEAD"));
+
+	cl_git_pass(git_reference_create(&reference, g_repo, name, &id, 0, message));
+
+	cl_git_pass(git_reflog_read(&reflog, g_repo, name));
+	cl_assert_equal_sz(1, git_reflog_entrycount(reflog));
+
+	entry = git_reflog_entry_byindex(reflog, 0);
+	cl_assert(git_oid_streq(&entry->oid_old, GIT_OID_HEX_ZERO) == 0);
+	cl_assert_equal_oid(&id, &entry->oid_cur);
+	cl_assert_equal_s(message, entry->msg);
+
+	git_reflog_free(reflog);
+	git_reference_free(reference);
+}
+
+
+void test_refs_reflog_messages__renaming_ref(void)
+{
+	git_reference *ref, *new_ref;
+
+	cl_git_pass(git_reference_lookup(&ref, g_repo, "refs/heads/master"));
+	cl_git_pass(git_reference_rename(&new_ref, ref, "refs/heads/renamed", false,
+									 "message"));
+
+	reflog_check_entry(g_repo, git_reference_name(new_ref), 0,
+		"a65fedf39aefe402d3bb6e24df4d4f5fe4547750",
+		"a65fedf39aefe402d3bb6e24df4d4f5fe4547750",
+		"foo@example.com", "message");
+
+	git_reference_free(ref);
+	git_reference_free(new_ref);
+}
+
+void test_refs_reflog_messages__updating_a_direct_reference(void)
+{
+	git_reference *ref, *ref_out, *target_ref;
+	git_oid target_id;
+	const char *message = "You've been logged, mate!";
+
+	git_reference_name_to_id(&target_id, g_repo, "refs/heads/haacked");
+	cl_git_pass(git_reference_lookup(&target_ref, g_repo, "refs/heads/haacked"));
+
+	cl_git_pass(git_reference_lookup(&ref, g_repo, "refs/heads/master"));
+
+	cl_git_pass(git_reference_set_target(&ref_out, ref, &target_id, message));
+
+	reflog_check_entry(g_repo, "refs/heads/master", 0,
+		"a65fedf39aefe402d3bb6e24df4d4f5fe4547750",
+		"258f0e2a959a364e40ed6603d5d44fbb24765b10",
+		NULL, message);
+
+	git_reference_free(target_ref);
+	git_reference_free(ref);
+	git_reference_free(ref_out);
+}
+
+#define NEW_BRANCH_NAME "new-branch-on-the-block"
+
+void test_refs_reflog_messages__creating_branches_default_messages(void)
+{
+	git_buf buf = GIT_BUF_INIT;
+	git_annotated_commit *annotated;
+	git_object *obj;
+	git_commit *target;
+	git_reference *branch;
+
+	cl_git_pass(git_revparse_single(&obj, g_repo, "e90810b8df3"));
+	cl_git_pass(git_commit_lookup(&target, g_repo, git_object_id(obj)));
+	git_object_free(obj);
```

Comment:
I'd prefer if you moved all freeing functions to the end. It makes it a lot easier to see if everything has been taken care of by just comparing the list of declarations and free's at the end (at least in most cases).